docs/contrib/keycloak.md

```diff
@@ -0,0 +1,148 @@
+    4. Add a client for REMS, by navigating to `Clients` on the left side of the page and clicking `Create`. Set the `Client Protocol` to `openid-connect`. Clients are allowed to authenticate users within the Keycloak realm that they are bound to. For this example, the Client ID will be `rems-client`.
+    5. Upon client creation, you will be redirected to `Clients` > `rems-client`. Put `*` in the `Valid Redirect URIs` field.
+    6. There is a variety of ways to authenticate the client itself with Keycloak. REMS uses the `Client ID` and a Keycloak-generated `Secret`. In Keycloak, navigate to `Clients` > `rems-client` if you are not already there. Set the `Access Type` to `confidential`. Then, navigate to the `Credentials` tab. Set `Client Authenticator` to `Client Id and Secret`. Click `Regenerate Secret` and copy the generated secret to your clipboard. For this example, the generated secret is `0ed4b9e0-5ff0-446a-ab7b-82dac465b9d5`.
+
+3. **Configuring REMS to Authenticate with Keycloak**
+    1. Modify your custom REMS config file (called `./services/rems/simple-config.edn` in this example). Tell REMS to use OIDC by modifying the `:authentication` parameter, and setting the values for some OIDC parameters specific to your keycloak instance. Remember that docker-compose will mount this file into your REMS container as `/rems/config/config.edn`.
+    ```
+    :authentication :oidc
+
+    ; The location of Keycloak's OIDC metadata for the realm to which REMS is client
+    :oidc-metadata-url "http://keycloak:8080/auth/realms/rems-idp/.well-known/openid-configuration"
+
+    ; REMS' client ID, as defined in Keycloak
+    :oidc-client-id "rems-client"
+
+    ; REMS' client secret, as generated in keycloak in step #8 above
+    :oidc-client-secret "0ed4b9e0-5ff0-446a-ab7b-82dac465b9d5"
+    ```
+
+4. **Spinning up REMS**
+    1. Add any other configuration that your REMS instance needs to your `simple-config.edn` file. For example, you will want to populate the `database-url` and `port` parameters.
+    ```
+    {:port 3000
+    :database-url "postgresql://db:5432/rems?user=rems&password=remspassword"
+    :search-index-path "/tmp/rems-search-index"
+    :authentication :oidc
+    :public-url "http://localhost:3001/"
+
+    :oidc-metadata-url "http://rp-keycloak:8080/auth/realms/rems-idp/.well-known/openid-configuration"
+
+    :oidc-client-id "rems-client"
+    :oidc-client-secret "0ed4b9e0-5ff0-446a-ab7b-82dac465b9d5"
+    }
+    ```
+    2. Build rems, which (at the time of writing) requires:
+        1. [Installing Leiningen](https://leiningen.org/)
+        2. Building the Leiningen jar: `lein uberjar`
+        3. Building the Docker container: `docker-compose build rems`
+        4. Migrating the test data: `docker-compose run --rm -e CMD="migrate;test-data" rems`
```

Comment:
I think the alternative we could use here is to point to the [Docker Hub REMS image](https://hub.docker.com/r/cscfi/rems).